larpandoracontent/LArVertex/TrainedVertexSelectionAlgorithm.cc

```diff
@@ -295,28 +296,49 @@ void TrainedVertexSelectionAlgorithm::GetEventShapeFeatures(const ClusterList &c
         this->UpdateSpanCoordinate(minPosition.GetZ(), maxPosition.GetZ(), zMin, zMax);
     }
 
-    const float xSpan(this->GetCoordinateSpan(xMax, xMin));
-    const float ySpan(this->GetCoordinateSpan(yMax, zMin));
-    const float zSpan(this->GetCoordinateSpan(yMax, zMin));
-
-    // Calculate the volume and longitudinality of the event (ySpan often 0 - to be investigated).
-    if ((xSpan > std::numeric_limits<float>::epsilon()) && (ySpan > std::numeric_limits<float>::epsilon()))
-    {
-        eventVolume     = xSpan * ySpan * zSpan;
-        longitudinality = zSpan / (xSpan + ySpan);
-    }
-
-    else if (ySpan > std::numeric_limits<float>::epsilon())
+    // Calculate the area and longitudinality of the event. Option retained for deprecated implementation.
+    if (m_use2DEventShapes)
     {
-        eventVolume     = ySpan * ySpan * zSpan;
-        longitudinality = zSpan / (ySpan + ySpan);
+      const float xSpan(this->GetCoordinateSpan(xMax, xMin));
+      const float zSpan(this->GetCoordinateSpan(zMax, zMin));
+      
+      if ((xSpan > std::numeric_limits<float>::epsilon()))
+      {
+	eventArea       = xSpan * zSpan;
+	longitudinality = zSpan / (xSpan + zSpan);
+      }
     }
-
-    else if (xSpan > std::numeric_limits<float>::epsilon())
+    
+    else
     {
-        eventVolume     = xSpan * xSpan * zSpan;
-        longitudinality = zSpan / (xSpan + xSpan);
-    }
+      if (m_trainingSetMode)
+      {
+	std::cout << "TrainedVertexSelectionAlgorithm: WARNING -- Using deprecated event shape features for training, "
```

Comment:
This warning will be printed a lot (maybe a good thing!), but could consider moving it to the ReadSettings method, so that it's printed clearly once, at the start of a training job, after the XML parsing has been done and configuration is fixed. We can also add a TODO comment here, next to the error, to help note that the legacy mode is to disappear once new trainings are available for the different use cases.